core/primitives-core/src/account.rs

```diff
@@ -154,6 +202,34 @@ impl Account {
         }
     }
 
+    #[inline]
+    pub fn global_contract_hash(&self) -> Option<CryptoHash> {
+        match self {
+            Self::V2(AccountV2 { contract: AccountContract::Global(hash), .. }) => Some(*hash),
+            Self::V1(_) | Self::V2(_) => None,
+        }
+    }
+
+    #[inline]
+    pub fn global_contract_account_id(&self) -> Option<&AccountId> {
+        match self {
+            Self::V2(AccountV2 { contract: AccountContract::GlobalByAccount(account), .. }) => {
+                Some(account)
+            }
+            Self::V1(_) | Self::V2(_) => None,
+        }
+    }
+
+    #[inline]
+    pub fn local_contract_hash(&self) -> Option<CryptoHash> {
+        match self.contract() {
```

Comment:
nit: in the current implementation calling `self.contract()` results in potentially copying `AccountId`, would be nice to address that somehow